CONTRIBUTING.rst

```diff
@@ -2,38 +2,56 @@
-
-These include:
+Klein is an open source project that welcomes contributions of all kinds coming from the community, including:
 
 - code patches,
 - `documentation <http://klein.readthedocs.org/>`_ improvements,
 - `bug reports <https://github.com/twisted/klein/issues>`_,
 - reviews for `contributed patches <https://github.com/twisted/klein/pulls>`_.
 
-Here's some suggestions to make the contributing process much smoother for you:
+
+Getting started
+===============
+
+Here is a list of shell commands that will install the dependencies of Klein, run the test suite, and compile the documentation.
+
+::
+   pip install --user -r requirements.txt
+   tox
+   tox -e docs
+
+
+Next steps
+----------
+
+Here are some suggestions to make the contributing process easier for everyone:
 
 Code
-====
+----
 
 - Propose all patches through a pull request in the `GitHub repo <https://github.com/twisted/klein>`_.
-- Use `Twisted's code style guide <http://twistedmatrix.com/documents/current/core/development/policy/coding-standard.html>`_ for all code you touch, as long as it doesn't break a public API.
-- Make sure your patch has tests, since untested code is buggy code.
+- Use `Twisted's coding standards <http://twistedmatrix.com/documents/current/core/development/policy/coding-standard.html>`_ as a guideline for code changes you make.
+- Code changes should have tests: untested code is buggy code.
   Klein uses `Twisted Trial <http://twistedmatrix.com/documents/current/api/twisted.trial.html>`_ and `tox <https://testrun.org/tox/latest/index.html>`_ for its tests.
   The command to run the full test suite is ``tox`` with no arguments.
   This will run tests against several versions of Python and Twisted, which can be time-consuming.
   To run tests against only one or a few versions, pass a ``-e`` argument with an environment from the envlist in ``tox.ini``: for example, ``tox -e py27-tw150`` will run tests with Python 2.7 and Twisted 15.0.
   To run only one or a few specific tests in the suite, add a filename or fully-qualified Python path to the end of the test invocation: for example, ``tox klein.test.test_app.KleinTestCase.test_foo`` will run only the ``test_foo()`` method of the ``KleinTestCase`` class in ``klein/test/test_app.py``.
   These two test shortcuts can be combined to give you a quick feedback cycle, but make sure to check on the full test suite from time to time to make sure changes haven't had unexpected side effects.
-- If it's a new feature, add an example into the `examples directory <https://github.com/twisted/klein/tree/master/docs/examples>`_ and add it to the index.
+- If you're adding a new feature, please add an example and some explanation to the `examples directory <https://github.com/twisted/klein/tree/master/docs/examples>`_, then add your example to the index.
+- Please run `flake8 <https://flake8.readthedocs.org/>`_ or a similar tool on any changed code.
```

Comment:
Klein is a weird mix of PEP8 and Twisted code, so flake8 may not help us. We do have a tox environment for pyflakes (the "flake" in flake8) which just does the checking for unused imports, etc -- it can be ran with `tox -e pyflakes`. Maybe this should be moved to "getting started" above.